lib/galaxy/util/custom_templates/template.py

```diff
@@ -0,0 +1,52 @@
+"""Render customizable templates e.g. emails with Jinja
+
+Templates are preferentially read from the ``config.templates_dir directory``.
```

Comment:
```suggestion
Templates are preferentially read from the ``templates_dir`` directory.
```
The parent directory (`config_dir`)  is already set in the schema, so we don't have to duplicate that here; otherwise we'd have to remember to modify it in both places if it moves. Also, to keep it consistent across Galaxy's docs and inline comments: we usually refer to those directories by their name only (e.g. `data_dir`, `tool_dependency_dir`) regardless of their parent.

Reply:
But if a custom `config_dir` is set, we won't be able to find the default templates any more, right? That's why I hard-coded that here, because the path to Galaxy's default templates should never change? Is there a better way to do this?

Reply:
i.e. we need to have access to both of these at once:
- `galaxy/config/templates`
and
- `whatever/my_config/custom_templates`

... since the latter might not include the requested template.

Reply:
if you were to place the templates into `[config_dir]/templates`, they will be available at that location no matter what `config_dir` is set to: it'll be the value of `app.config.config_dir`. The ***relative*** path to templates (or anything else) should never change, but it is resolved to an absolute path with respect to its parent's path, whatever that path is.

Also, defaults and custom files can live in the same location - more on this in a separate comment.

Reply:
On second thought, you're right - with a custom config, defaults will be unreachable! In fact, default templates don't have to be in the `config` directory: it's just a fallback value, and as such it's part of the code base, not configuration! So, here's an idea: you could place the defaults in `lib/galaxy/config/templates/mail/` (other "non-mail" templates could go into the parent or other subdirectories as needed). Then you do not need to git-ignore the `galaxy-root/config` directory (contrary to my initial suggestion). Then the only place in the code that knows about the location of the default templates is your util module. Will this work?

Reply:
Yep that sounds great :slightly_smiling_face: Although it might be good to keep them in the `config_dir` just as `.sample` files? That way it's easy for people to copy/tweak them? I guess I should probably include a `make config-templates` to build the samples from the defaults, if that's the case?

Reply:
You could just make symlinks to the defaults - like the sample config files (e.g. `config/galaxy.yml.sample`)